gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagManager.java

```diff
@@ -470,7 +531,18 @@ private void submitJob(DagNode<JobExecutionPlan> dagNode) {
         // The SpecProducer implementations submit the job to the underlying executor and return when the submission is complete,
         // either successfully or unsuccessfully. To catch any exceptions in the job submission, the DagManagerThread
         // blocks (by calling Future#get()) until the submission is completed.
-        producer.addSpec(jobSpec).get();
+        Future addSpecFuture = producer.addSpec(jobSpec);
+
+        if (addSpecFuture instanceof CompletableFuture
```

Comment:
If this block is only about setting azexecId, plz wrap it in a function

Reply:
done